quickwit/quickwit-storage/src/object_storage/s3_compatible_storage.rs

```diff
@@ -747,6 +767,23 @@ impl Storage for S3CompatibleObjectStorage {
             })
     }
 
+    #[instrument(level = "debug", skip(self, range), fields(range.start = range.start, range.end = range.end))]
+    async fn get_slice_stream(
+        &self,
+        path: &Path,
+        range: Range<usize>,
+    ) -> crate::StorageResult<Box<dyn AsyncRead + Send + Unpin>> {
+        let permit = REQUEST_SEMAPHORE.acquire().await;
+        let get_object_output = retry(&self.retry_params, || {
+            self.create_get_object_request(path, Some(range.clone()))
+        })
+        .await?;
+        Ok(Box::new(S3AsyncRead {
+            read: get_object_output.body.into_async_read(),
```

Comment:
yes! but it implies more changes in the FileSource which does expect an `AsyncRead` because it can also use stdin when no path is specified: https://github.com/quickwit-oss/quickwit/blob/46e65f8a6cc5cb3ad5bfd6238b27d54fcbaf7dd5/quickwit/quickwit-indexing/src/source/file_source.rs#L141-L156

Reply:
Understood!